lib/position/bk.js

```diff
@@ -211,36 +211,69 @@ function horizontalCompaction(g, layering, root, align, reverseSep) {
   // coordinates. The second sweep removes unused space by moving blocks to the
   // greatest coordinates without violating separation.
   var xs = {},
-      blockG = buildBlockGraph(g, layering, root, reverseSep);
-
-  // First pass, assign smallest coordinates via DFS
-  var visited = {};
-  function pass1(v) {
+      visited = {},
+      blockG = buildBlockGraph(g, layering, root, reverseSep),
+      borderType = reverseSep ? "borderLeft" : "borderRight";
+
+  // This function handles iterating over the graph without adding functions
+  // calls to the stack. The argument, helper, returns an array of functions.
+  // These returned functions are the recursive and base function calls. By
```

Comment:
Consider renaming "helper" to something more descriptive... maybe something like getNextStepTraversalFuncs . Thoughts?

Reply:
Done.